src/credentials/CHIPCert.cpp

```diff
@@ -1427,16 +1433,26 @@ CHIP_ERROR CertificateValidityPolicy::ApplyDefaultPolicy(const ChipCertificateDa
     }
 }
 
-CHIP_ERROR ValidateChipNetworkIdentity(const ByteSpan & cert)
+void InitNetworkIdentitySubject(ChipDN & name)
 {
-    ChipCertificateData certData;
-    ReturnErrorOnFailure(DecodeChipCert(cert, certData, CertDecodeFlags::kGenerateTBSHash));
+    name.Clear();
+    CHIP_ERROR err = name.AddAttribute_CommonName(kNetworkIdentityCN, /* not printable */ false);
+    VerifyOrDie(err == CHIP_NO_ERROR); // AddAttribute can't fail in this case
+}
 
-    CertType certType;
-    ReturnErrorOnFailure(certData.mSubjectDN.GetCertType(certType));
-    VerifyOrReturnError(certType == CertType::kNetworkIdentity, CHIP_ERROR_WRONG_CERT_TYPE);
+static void CalculateKeyIdentifierSha256(const P256PublicKeySpan & publicKey, MutableCertificateKeyId keyId)
+{
+    uint8_t hash[kSHA256_Hash_Length];
+    static_assert(keyId.size() <= sizeof(hash)); // truncating 32 bytes down to 20
+    chip::Crypto::Hash_SHA256(publicKey.data(), publicKey.size(), hash);
```

Comment:
Need to handle error here, since it can fail on some platforms without some pre-inits. You could VerifyOrDie, but not sure it's the safest.

Reply:
Ah right, must have missed the CHIP_ERROR return value. I wonder if we should declare ChipError `[[nodiscard]]`. Results in tons of errors currently though.